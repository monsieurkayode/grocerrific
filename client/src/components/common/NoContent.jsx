import React from 'react';
import { string } from 'prop-types';

const NoContent = ({ content }) => (
  <div className="no-content">
    <i className="material-icons">info</i>
    <h3 className="no-content-info">
      {content}
    </h3>
  </div>
);

NoContent.propTypes = {
  content: string.isRequired
};

export default NoContent;
