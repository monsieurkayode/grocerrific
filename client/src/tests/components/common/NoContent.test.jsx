import React from 'react';
import { shallow } from 'enzyme';
import NoContent from '../../../components/common/NoContent';

describe('NoContent Component', () => {
  const wrapper = shallow(<NoContent content="404! Not Found" />);

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('has a single root div with class no-content', () => {
    expect(wrapper.find('div').length).toBe(1);
    expect(wrapper.find('.no-content').length).toBe(1);
  });

  it('has a h3 element with text content 404! Not Found', () => {
    expect(wrapper.find('h3').length).toBe(1);
    expect(wrapper.find('h3').text()).toBe('404! Not Found');
  });
});
