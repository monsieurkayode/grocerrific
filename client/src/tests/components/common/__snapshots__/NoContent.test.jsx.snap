// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoContent Component renders correctly 1`] = `
<div
  className="no-content"
>
  <i
    className="material-icons"
  >
    info
  </i>
  <h3
    className="no-content-info"
  >
    404! Not Found
  </h3>
</div>
`;
